php/src/vaas/Vaas.php

```diff
@@ -63,21 +66,21 @@ public function Connect(
         string $token,
         ?VaasConnection $vaasConnection = null
     ) {
-        if (isset($vaasConnection)) {
-            $this->_vaasConnection = $vaasConnection;
-        } else {
-            $this->_vaasConnection = new VaasConnection($this->_vaasUrl);
-        }
+        $this->_vaasConnection = isset($vaasConnection) ? $vaasConnection :  new VaasConnection($this->_vaasUrl);
         $webSocket = $this->_vaasConnection->GetConnectedWebsocket();
 
         $authRequest = new AuthRequest($token);
         $webSocket->send(json_encode($authRequest));
         $authResponse = $this->_waitForAuthResponse();
-        if ($this->_logger != null)
-            $this->_logger->debug("Authenticated: " . json_encode($authResponse));
+        $this->log("debug", "Authenticated: " . json_encode($authResponse));
         $this->_vaasConnection->SessionId = $authResponse->session_id;
     }
 
+    private function log($level, string|Stringable $message, array $context = []): void {
```

Comment:
Mit dem NullLogger kann das raus:

http://phergie.github.io/api/classes/Psr.Log.NullLogger.html



This Logger can be used to avoid conditional log calls

Logging should always be optional, and if no logger is provided to your library creating a NullLogger instance to have something to throw logs at is a good way to avoid littering your code with if ($this->logger) { } blocks.